--- a/customtreectrl.py	2012-11-18 15:11:07.000000000 +0100
+++ b/customtreectrl.py	2012-11-18 15:08:42.000000000 +0100
@@ -1601,6 +1601,7 @@
 
         self._isCollapsed = True
         self._hasHilight = False    # same as focused
+        self._hilightPart = 0       # 0: full; else wx.TREE_HITTEST_ONITEM[LOWER|UPPER]PART
         self._hasPlus = False       # used for item which doesn't have
                                     # children but has a [+] button
         self._isBold = False        # render the label in bold font
@@ -2098,9 +2099,20 @@
         :param bool `set`: ``True`` to set the focus to the item, ``False`` otherwise.    
         """
 
-        self._hasHilight = set 
+        self._hasHilight = set
+        self._hilightPart = 0
 
 
+    def SetPartialHilight(self, flags=0):
+        """
+        Sets the item hilight style.
+
+        :param `flags`: ``0`` to hilight all of the item; ``wx.TREE_HITTEST_ONITEMUPPERPART`` to hilight the upper third or ``wx.TREE_HITTEST_ONITEMLOWERPART`` for the lower third.
+        """
+
+        self._hilightPart = flags
+        
+
     def HasChildren(self):
         """
         Returns whether the item has children or not.
@@ -2120,6 +2132,12 @@
 
         return self._hasHilight != 0 
 
+    def PartialHilight(self):
+        """
+        Returns the hilight style.
+        """
+
+        return self._hilightPart
 
     def IsExpanded(self):
         """
@@ -7125,7 +7143,7 @@
                 event.Skip()
                 return
             
-            if not event.HasModifiers():
+            if not (event.HasModifiers() or event.CmdDown()):
                 event = TreeEvent(wxEVT_TREE_ITEM_ACTIVATED, self.GetId())
                 event._item = self._current
                 event.SetEventObject(self)
