--- agw-old/hypertreelist.py	2012-10-14 17:49:22.000000000 +0200
+++ agw-new/hypertreelist.py	2013-02-24 15:55:31.000000000 +0100
@@ -4138,9 +4138,16 @@
         self._header_win = None
         self._main_win = None
         self._headerHeight = 0
         self._attr_set = False
-        
+
+        if '__WXMAC__' in wx.PlatformInfo:
+            self._minHeaderHeight = 17
+        elif '__WXMSW__' in wx.PlatformInfo:
+            self._minHeaderHeight = 20
+        else:
+            self._minHeaderHeight = 26
+
         main_style = style & ~(wx.SIMPLE_BORDER|wx.SUNKEN_BORDER|wx.DOUBLE_BORDER|
                                wx.RAISED_BORDER|wx.STATIC_BORDER)
 
         self._agwStyle = agwStyle
@@ -4175,9 +4182,9 @@
     def CalculateAndSetHeaderHeight(self):
         """ Calculates the best header height and stores it. """
 
         if self._header_win:
-            h = wx.RendererNative.Get().GetHeaderButtonHeight(self._header_win)
+            h = max(wx.RendererNative.Get().GetHeaderButtonHeight(self._header_win), self._minHeaderHeight)
             # only update if changed
             if h != self._headerHeight:
                 self._headerHeight = h
                 self.DoHeaderLayout()
